src/app/patient/myappointments/service/myapp.service.ts

```diff
@@ -20,4 +21,11 @@ export class MyappService {
   cancelAppointment(id : number): Observable<IAppointment[]> {
     return this.http.post<IAppointment[]>(this.apiHost + 'api/public/appointments/cancel',id, {headers: this.headers});
   }
+
+  showPdf(appointmentId : number): Observable<IExam> {
+    return this.http.post<IExam>(this.apiHost + 'api/public/appointments/pdf',appointmentId, {headers: this.headers});
+  }
+  getAllFinishedAppointments(): Observable<IAppointment[]> {
```

Comment:
There is mostly no need to put method type, because typescript has automatic type inference.